### DISCUSSION TITLE
How to generate power variogram?

### DISCUSSION THREAD

**Discussion by Martin20494** [GeoStat-Framework/GSTools#285] [Q&A]
Hi team,

Thank you for a great package! I have a question about variogram/covariance model. Can gstools generate power variogram (as [here](https://www.researchgate.net/figure/Some-types-of-variograms-Three-types-of-variogram-models-apply-to-different-situations_fig2_254560058))? (Sorry I also had the same question on gitter)

**Comment by MuellerSeb** [comment]
Hi there,

Power variogram are not directly supported, since we only support bounded variogram models. (There is an open issue for unbounded models: https://github.com/GeoStat-Framework/GSTools/issues/192)

BUT: A special feature of GSTools are [truncated power law variograms](https://geostat-framework.readthedocs.io/projects/gstools/en/v1.4.0/examples/00_misc/00_tpl_stable.html). These variograms are an approximation of power law variogram with an upper cut of bound. You could have a look there, if that already helps you.

See the publication by [Di Federico and Neuman (1997)](https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/97WR00299) for theoretical background.

Hope that helps. Cheers,
Sebastian

**Reply by Martin20494** [reply]
Hi Mueller,

Thank you so much for your reply.

I have tried the truncated power law variograms, but unfortunately, it does not have the same shape as my estimated variogram to fit. I attached the code and the results as below. Do you still have any other suggestion?

```
# Package
import pandas as pd
import gstools as gs

# Read file
file_points = pd.read_csv(<path to file>, index_col=0)

# Estimate variogram
bin_center, gamma = gs.vario_estimate((file_points.x, file_points.y), file_points.z)

# Fit variogram
var = gs.TPLStable(
    dim=2,
    var=3,
    len_low=0,
    len_scale=1750,
    nugget=0.0001,
    hurst=0.5
)

var.fit_variogram(bin_center, gamma)

# Plot original variogram
ax = var.plot(x_max=max(bin_center))
ax.scatter(bin_center, gamma)
```

![power_vario](https://user-images.githubusercontent.com/55137629/208277858-416c0283-2c68-47f7-98bf-385a133b0e0e.png)
